### ISSUE TITLE
no-useless-rename triggered when there's no rename

### ISSUE THREAD

**Issue by daltones** [eslint/eslint#6266]
**What version of ESLint are you using?**
v2.11.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default (babel-eslint shows the same issue)

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "rules": {
    "no-useless-rename": 1
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
import Foo from 'foo';
import {bar as otherBar} from 'bar';
import {fooBar} from 'foo-bar';
import {baz as baz} from 'baz';
```

**What did you expect to happen?**
Just one warning in line 4.

**What actually happened? Please include the actual, raw output from ESLint.**
```
/path-to-the-file/index.js
  3:9  warning  Import fooBar unnecessarily renamed  no-useless-rename
  4:9  warning  Import baz unnecessarily renamed     no-useless-rename

✖ 2 problems (0 errors, 2 warnings)
```

**Comment by kaicataldo** [comment]
Thanks for the issue. Not sure what's going on, but I'll look into this when I'm back at my computer.

**Comment by alberto** [comment]
Confirmed.

**Comment by kaicataldo** [comment]
Figured this out. Oversight on my part when I wrote his rule - fix incoming.

**Comment by kaicataldo** [comment]
Oh, just saw @alberto's got this covered. Am on mobile and missed it. Thanks - somehow completely missed this in my tests (sorry!).

**Comment by daltones** [comment]
Thanks for the fast replies! :)

**Comment by daltones** [comment]
I've just upgraded to 2.11.1 and the problem was fixed for _default_ parser.

But it is still there when using **babel-eslint**.

**Comment by kaicataldo** [comment]
babel-eslint parses the code differently than the default parser and can cause some rules to not work. Consider creating an issue directly with babel-eslint to see if this can be resolved either in the parser itself or in a plugin like https://github.com/babel/eslint-plugin-babel.

**Comment by alberto** [comment]
Mmmm, while what @kaicataldo says is generally true, I think in this case I was relying on a parser internal (reference equality), so we should probably fix that. Sorry, my fault.

PR is coming.
